packages/fxa-settings/src/lib/gql.ts

```diff
@@ -8,27 +8,19 @@ import { ErrorHandler, onError } from '@apollo/client/link/error';
 import { BatchHttpLink } from '@apollo/client/link/batch-http';
 import { cache, sessionToken, typeDefs } from './cache';
 
-const pagesViewableWithoutAuthentication = [
-  'primary_email_verified',
-  'primary_email_confirmed',
-  'reset_password_verified',
-  'reset_password_with_recovery_key_verified',
-  'signup_verified',
-  'signup_confirmed',
-  'signin_confirmed',
-  'signin_verified',
-];
+export const pagesRequiringAuthentication = ['settings'];
 
 export const errorHandler: ErrorHandler = ({ graphQLErrors, networkError }) => {
   let reauth = false;
 
-  const currentPageDoesNotRequireAuthentication =
-    pagesViewableWithoutAuthentication.some((urlSnippet) => {
-      // We check if the url for the current page contains the path of a page which does not require authentication
+  const currentPageRequiresAuthentication = pagesRequiringAuthentication.some(
+    (urlSnippet) => {
+      // We check if the url for the current page contains the path of a page which requires authentication
       return (
         window?.location?.href && window.location.href.includes(urlSnippet)
       );
```

Comment:
I guess this is technically existing behavior, but I'm not clear on how it works.

If I go to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat?foo=bar#baz and then use DevTools console to log my `window.location.href`, I get that full URL back.
So `String#includes()` will search for the `urlSnippet` substring _anywhere_ in that URL. 

Q: Should this be restricted to the `pathname` or something? Wouldn't this also consider a page as needing authentication if a query param or `hash` included `"settings"` anywhere in the string?
Or at the very least should `['settings']` above be more specific, like `['/settings']` to make it less likely to match unwanted strings?

```js
JSON.stringify(window.location, null, 2)
{
  "ancestorOrigins": {},
  "hash": "#baz",
  "host": "developer.mozilla.org",
  "hostname": "developer.mozilla.org",
  "href": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat?foo=bar#baz",
  "origin": "https://developer.mozilla.org",
  "pathname": "/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat",
  "port": "",
  "protocol": "https:",
  "search": "?foo=bar"
}
```

Reply:
That's a great idea!

Reply:
Fixed!